dashboard/src/actions/repos.ts

```diff
@@ -248,6 +256,8 @@ export const updateRepo = (
         registrySecrets,
       );
       dispatch(repoUpdated(data.appRepository));
+      // Re-fetch secrets in case there are updates
+      dispatch(fetchRepoSecrets(namespace));
```

Comment:
So given the different secret associated with an AppRepository - helm repository secrets and docker registry image pull secrets, this could be confusing? Perhaps a more explicit comment:

```
// Re-fetch the helm repo secret that could have been modified with the updated headers
// so that if the user chooses to edit the app repo again, they will see the current value.
```
if I have understood correctly? Also, don't we expect only ever *one* secret for the helm repository access? If so, why aren't we encoding that anywhere? Finally, would the code be simpler if the AppRepository actually stored in a field the name of the helm repository secret?

Reply:
we can request just one secret, yes